webpack/assets/javascripts/react_app/components/common/emptyState/index.js

```diff
@@ -0,0 +1,70 @@
+import React from 'react';
+import { EmptyState as PfEmptyState, Button } from 'patternfly-react';
+import API from '../../../API';
+import helper from '../../../common/helpers';
+import { simpleLoader } from '../../common/Loader';
+
+class EmptyState extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      data: undefined,
+    };
+  }
+
+  componentDidMount() {
+    const { data, uri, model } = this.props;
+    const url = model
+      ? helper.urlBuilder(this.props.model, 'empty_state')
+      : uri;
+
+    if (data) {
+      this.setState({ data });
+    } else {
+      API.get(url).then(
+        result => this.setState({ data: result.data }),
+        (jqXHR, textStatus, error) => this.setState({ error: true }),
+      );
+    }
+  }
+  render() {
+    if (!this.state.data) {
+      return simpleLoader('md');
+    }
+    const {
+      icon,
+      header,
+      description,
+      docUrl,
+      action,
+      secondayActions,
+    } = this.state.data;
+
+    return (
+      <PfEmptyState>
+        <PfEmptyState.Icon type="pf" name={icon || 'add-circle-o'} />
+        <PfEmptyState.Title>{header}</PfEmptyState.Title>
+        <PfEmptyState.Info>{__(description)}</PfEmptyState.Info>
+        <PfEmptyState.Help>
+          For more information please see{' '}
+          <a href={docUrl}>{__('documentation')}</a>
```

Comment:
I'm afraid this is going to cause issues with translating languages with different word order. It effectively prevents from putting the word "documentation" anywhere else in the sentence.

Reply:
there is also https://github.com/theforeman/foreman/blob/develop/webpack/assets/javascripts/react_app/components/common/DocumentationLink/index.js maybe we can combine it somehow?